src/nativeScriptHashBuilder.c

```diff
@@ -0,0 +1,229 @@
+#include "cbor.h"
+#include "nativeScriptHashBuilder.h"
+
+//#define TRACE_NATIVE_SCRIPT_HASH_BUILDER
+
+#ifdef TRACE_NATIVE_SCRIPT_HASH_BUILDER
+#define _TRACE(...) TRACE(__VA_ARGS__)
+#define _TRACE_BUFFER(...) TRACE_BUFFER(__VA_ARGS__)
+#else
+#define _TRACE(...)
+#define _TRACE_BUFFER(...)
+#endif // TRACE_NATIVE_SCRIPT_HASH_BUILDER
+
+#define APPEND_CBOR(type, value) blake2b_224_append_cbor_data(&builder->nativeScriptHash, type, value)
+#define APPEND_BUFFER(buffer, size) blake2b_224_append_buffer_data(&builder->nativeScriptHash, buffer, size)
+
+static void blake2b_224_append_buffer_data(
+        blake2b_224_context_t* hashCtx,
+        const uint8_t* buffer, size_t size
+)
+{
+	_TRACE_BUFFER(buffer, size);
+	blake2b_224_append(hashCtx, buffer, size);
+}
+
+__noinline_due_to_stack__
+static void blake2b_224_append_cbor_data(
+        blake2b_224_context_t* hashCtx,
+        uint8_t type, uint64_t value
+)
+{
+	uint8_t buffer[10];
+	size_t size = cbor_writeToken(type, value, buffer, SIZEOF(buffer));
+	_TRACE_BUFFER(buffer, size);
+	blake2b_224_append(hashCtx, buffer, size);
+}
+
+static inline void advanceState(native_script_hash_builder_t* builder)
+{
+	// advance state should be only called when state is not finished
+	// the advance state determines the next state from the current level
+	// and number of remaining scripts for the current level
+	ASSERT(builder->state == NATIVE_SCRIPT_HASH_BUILDER_SCRIPT);
+
+	_TRACE("Advancing state, level = %u, remaining scripts = %u", builder->level, builder->remainingScripts[builder->level]);
+
+	if (builder->level == 0 && builder->remainingScripts[builder->level] == 0) {
+		builder->state = NATIVE_SCRIPT_HASH_BUILDER_FINISHED;
+	} else {
+		builder->state = NATIVE_SCRIPT_HASH_BUILDER_SCRIPT;
+	}
+}
+
+static inline bool isComplexScriptFinished(native_script_hash_builder_t* builder)
+{
+	return builder->level > 0 && builder->remainingScripts[builder->level] == 0;
+}
+
+static void complexScriptFinished(native_script_hash_builder_t* builder)
+{
+	while(isComplexScriptFinished(builder)) {
```

Comment:
```suggestion
	while (isComplexScriptFinished(builder)) {
```
Please check for missing spaces in other similar places (e.g. `if(`).

And `make format` needs to be run.

Reply:
If possible when I will be cleaning up the commit history I will run `make format` for each of the commits and resolve this issue then, if it's okay with you.

Reply:
Yes, I'm fine with that.

Reply:
This was actually not caught and fixed by `make format`